server/src/main/java/org/opensearch/index/shard/IndexShard.java

```diff
@@ -587,7 +587,7 @@ public boolean shouldSeedRemoteStore() {
     public Function<String, Boolean> isShardOnRemoteEnabledNode = nodeId -> {
         DiscoveryNode node = discoveryNodes.get(nodeId);
         if (node != null) {
-            return node.isRemoteStoreNode();
+            return node.isRemoteSegmentStoreNode();
```

Comment:
This is mostly used for also tracking the dual replication flow during migration from docrep to remote store. How would the behaviour be for docrep to remote segment only migration? Also, lets double check that we are not loosing any invariant check for existing flow. 
cc @gbbafna @shourya035

Reply:
If this is only for the migration case, then I can switch it back.